### ISSUE TITLE
Use consistant time elements

### ISSUE THREAD

**Issue by jordan2175** [STIXProject/schemas#283]
Some times we use @timestamp, some times @timestamp and @timestamp_precision, and some times DateTimePrecisonType

**Comment by johnwunder** [comment]
To deconflict these usages a little:

#### `@timestamp` without `@timestamp_precision`
This only occurs in places where we use `@timestamp` for versioning. The reasoning behind not adding a precision is that because it's used for versioning and will be created by tools you would never have a need to specify precision.

#### `DateTimeWithPrecisionType`
This is our preferred approach to places that DO require a precision associated with a time. It adds an attribute called `@precision` to the DateTime element in order to do that. Unfortunately, it only works in places where the timestamp is an element. You obviously can't add attributes to other attributes so it doesn't work in places where the existing field was already an attribute.

#### `@timestamp` and `@timestamp_precision`
This is used in places where we need a precision but can't use `DateTimeWithPrecisionType` because the value is an attribute. The intent is that in a major release of STIX we would convert all of these attributes into elements and therefore you would not need `@timestamp_precision` anymore (all usages would be elements w/ `DateTimeWithPrecisionType`)

So to summarize, as of now the intend is to consolidate the latter two into a single approach via DateTimeWithPrecisionType. As of now we've not identified a use-case for precision on versioning timestamps and so I think those would still stay separate.

I don't think we had a ticket for adding that consolidation though so thanks for pointing it out! We can now use this as the issue to track that activity.

**Comment by jordan2175** [comment]
If we are going to use timestamp for version aspects, then we should call it that so people do not get confused.  versionTimestamp or something like it.  Or better yet, create a new element called version and add a version value and a timestamp value to it.  So it becomes really clear as to what is what. 

I would also like to see us get away from a lot of the attributes on elements.
